hi-michelleprimiani/pixo-api
# Cart - Create & Destroy Method

## Backend Support for Shopping Cart

### Description
This pull request introduces backend support for the shopping cart feature. It provides the necessary API endpoints to manage cart items, including adding new items to a cart, removing items, and fetching all items within a user's cart.

### Key Features:
**Add to Cart Endpoint:** POST /cartitems allows items to be added to the user's cart.
**Remove from Cart Endpoint:** DELETE /cartitems/<id> allows a specific item to be removed from the cart by its ID.
**List Cart Items Endpoint:** GET /cart returns all items in the user's current cart.
**Cart Model:** A new Cart and CartItem model to store user cart data and item details in the database.

## How To Test
Use the POST /cartitems endpoint to add a new item to the cart. Verify that the item is added correctly.
Use the DELETE /cartitems/<id> endpoint to remove an item from the cart. Verify that the item is removed and the cart total updates.
Use the GET /cart endpoint to list all items in the cart. Verify that all items and the total price are returned correctly.
